src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -231,11 +231,16 @@ private async void ProcessIncomingFrames()
             }
         }
 
-        // Note, this will return null for a streamId that's not in use.
+        // Note, this will return null for a streamId that's no longer in use.
         // Callers must check for this and send a RST_STREAM or ignore as appropriate.
+        // If the streamId is invalid or the stream is idle, calling this function
+        // will result in a connection level error.
```

Comment:
Per RFC (in PR description):

> Receiving any frame other than HEADERS or PRIORITY on a stream in this (idle) state MUST be treated as a connection error (Section 5.4.1) of type PROTOCOL_ERROR.

What will happen if a `HeadersFrame` is received for a idle stream? Also throw a connection error?

Reply:
Because seems after the change, we will throw `Http2ProtocolException` in this case, right?

Reply:
That's a good question! I talked to @geoffkizer about this online, and the basic conclusion is "it's ambiguous". It isn't really a meaningful operation for a server to send unsolicited headers to the client on an idle stream. Given that, I decided to go with the behavior set out for other types of unexpected frames:
>In the absence of more specific guidance elsewhere in this document, implementations SHOULD treat the receipt of a frame that is not expressly permitted in the description of a state as a connection error (Section 5.4.1) of type PROTOCOL_ERROR.

That behavior is pinned in the test `HeadersFrame_IdleStream_ConnectionError`, with a disclaimer that the spec is ambiguous here.